[DPROT-183, DPROT-217, DPROT-221, DVCSMP-2250] Update dths zigbee library

varzac opened the pull request:
This includes changes that address:

https://smartthings.atlassian.net/browse/DPROT-183
https://smartthings.atlassian.net/browse/DPROT-217
https://smartthings.atlassian.net/browse/DPROT-221
https://smartthings.atlassian.net/browse/DVCSMP-2250

@tpmanley

varzac committed:
Update DTHs to use zigbee library

This change updates the core DTHs to use the
zigbee library where appropriate instead of having its own zigbee
command strings.

varzac committed:
Clean up smartsense DTHs

This continues an effort to clean up the SmartSense DTHs and move as
much of the logic as possible into the library.  This simplifies the DTH
and has the advantage of having only a single location where issues need
to be fixed.

tpmanley commented:
@workingmonk be good to get your feedback on this too

tpmanley commented on devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy:
```diff
@@ -60,7 +61,7 @@ def updated() {
 
 def configure() {
-    def cmds = zigbee.configureReporting(0x0001, 0x0020, 0x20, 20, 20, 0x01)
+    def cmds = zigbee.configureReporting(0x0001, 0x0020, DataType.UINT8, 20, 20, 0x01)
```
Can use `zigbee.batteryConfig` instead

tpmanley commented on devicetypes/smartthings/smartpower-outlet.src/smartpower-outlet.groovy:
```diff
@@ -152,11 +153,7 @@ def configure() {
-		"zcl global send-me-a-report 0x0B04 0x050B 0x29 1 600 {05 00}",				//The send-me-a-report is custom to the attribute type for CentraLite
-		"send 0x${device.deviceNetworkId} 1 ${endpointId}", "delay 500"
-	]
+	zigbee.configureReporting(0x0B04, 0x050B, DataType.INT16, 1, 600, 0x0005)
```
Could use `zigbee. electricMeasurementPowerConfig`

tpmanley commented on devicetypes/smartthings/smartsense-multi-sensor.src/smartsense-multi-sensor.groovy:
```diff
@@ -328,25 +251,6 @@ private Map getBatteryResult(rawValue) {
-		def v = value as int
-		value = v + offset
-	}
-	def descriptionText = temperatureScale == 'C' ? '{{ device.displayName }} was {{ value }}°C':
```
I think we probably don't want to lose the descriptionText and `translatable: true` on the events. You could check the event map returned by `getEvent` and add these fields to the map.

tpmanley commented:
Otherwise it looks good

varzac committed:
Update to use multi attr parsing in zigbee library

With the addition of the ability for parseDescriptionAsMap to hanle the
return of multiple read attribute responses, we can be a little more
explicit with how we interpret and parse the responses.

workingmonk commented on devicetypes/smartthings/smartpower-dimming-outlet.src/smartpower-dimming-outlet.groovy:
```diff
@@ -69,288 +69,32 @@ metadata {
-		log.debug parseDescriptionAsMap(description)
+		log.debug zigbee.parseDescriptionAsMap(description)
+	} else if (event.name == "power") {
+		event.value = event.value / 10
```
Do you want to keep the reasoning on why we divide by 10?

workingmonk commented on devicetypes/smartthings/smartpower-outlet.src/smartpower-outlet.groovy:
```diff
@@ -45,80 +44,62 @@ metadata {
-			*/
+	if (event) {
+		if (event.name == "power") {
+			event.value = event.value / 10
```
reasoning for dividing by 10

tpmanley replied:
You could use the same comment here that you have in smartpower-dimming-outlet.groovy

workingmonk commented on devicetypes/smartthings/smartsense-multi-sensor.src/smartsense-multi-sensor.groovy:
```diff
@@ -73,210 +74,169 @@ metadata {
-		map = parseIasMessage(description)
-	}
+	def maps = []
+	maps << zigbee.getEvent(description)
```
can you make it consistent here? by doing `def maps = zigbee.getEvent(description)`

tpmanley replied:
I think this can be simplified a bit but since this version has already passed QA I think it is OK to release as is and clean up further in a separate PR.

workingmonk commented on devicetypes/smartthings/smartsense-multi-sensor.src/smartsense-multi-sensor.groovy:
```diff
@@ -73,210 +74,169 @@ metadata {
-	}
+	def maps = []
+	maps << zigbee.getEvent(description)
+	if (!maps[0]) {
```
`if(maps)`

workingmonk replied:
and all the `maps[0]` can go away.

workingmonk commented on devicetypes/smartthings/smartsense-multi-sensor.src/smartsense-multi-sensor.groovy:
```diff
@@ -516,25 +386,3 @@ private getManufacturerCode() {
 private hexToInt(value) {
```
if have a equivalent in coreApi. you can directly use `hexToIntConverter`
